presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestInlineProjections.java

```diff
@@ -0,0 +1,98 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.planner.assertions.ExpressionMatcher;
+import com.facebook.presto.sql.planner.assertions.PlanMatchPattern;
+import com.facebook.presto.sql.planner.iterative.rule.test.RuleTester;
+import com.facebook.presto.sql.planner.plan.Assignments;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.project;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.expression;
+import static io.airlift.testing.Closeables.closeAllRuntimeException;
+
+public class TestInlineProjections
+{
+    private RuleTester tester;
+
+    @BeforeClass
+    public void setUp()
+    {
+        tester = new RuleTester();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        closeAllRuntimeException(tester);
+        tester = null;
+    }
+
+    @Test
+    public void test()
+    {
+        tester.assertThat(new InlineProjections())
+                .on(p ->
+                        p.project(
+                                Assignments.builder()
+                                        .put(p.symbol("m", BIGINT), expression("x")) // identity
+                                        .put(p.symbol("n1", BIGINT), expression("y + 1")) // complex expression referenced multiple times
+                                        .put(p.symbol("n2", BIGINT), expression("y + 2")) // complex expression referenced multiple times
+                                        .put(p.symbol("o1", BIGINT), expression("z + 1")) // literal referenced multiple times
+                                        .put(p.symbol("o2", BIGINT), expression("z + 2")) // literal referenced multiple times
+                                        .put(p.symbol("p", BIGINT), expression("try(y / z)"))
+                                        .build(),
+                                p.project(Assignments.builder()
+                                                .put(p.symbol("x", BIGINT), expression("a"))
+                                                .put(p.symbol("y", BIGINT), expression("a * 2"))
```

Comment:
can you please add one complex expression which is then used only once, so we could see that it was inlined?